src/panels/developer-tools/state/developer-tools-state.js

```diff
@@ -297,8 +303,17 @@ class HaPanelDevState extends EventsMixin(LocalizeMixin(PolymerElement)) {
         return hass.states[key];
       })
       .filter(function (value) {
-        if (!value.entity_id.includes(_entityFilter.toLowerCase())) {
-          return false;
+        // If we have a match for the entity ID, we do not have to look further
+        const entityFilter = _entityFilter.toLowerCase();
+        if (!value.entity_id.includes(entityFilter)) {
+          // Entity ID does not match => check the friendly_name => if there is none,
+          // this entity is not relevant for rendering => return false
+          if (
+            value.attributes.friendly_name === undefined ||
+            !value.attributes.friendly_name.toLowerCase().includes(entityFilter)
+          ) {
+            return false;
+          }
```

Comment:
```suggestion
        if (!value.entity_id.includes(entityFilter) && value.attributes.friendly_name && !value.attributes.friendly_name.toLowerCase().includes(entityFilter)) {
            return false;
        }
```

Reply:
Your logic does not work to hide entities without friendly names when the filter does not match their entity ID. But I condensed my corrected logic to look more like your proposal.